### DISCUSSION TITLE
Automatically update based on new releases of APBS/PDB2PQR

### DISCUSSION THREAD

**Discussion by intendo** [Electrostatics/apbs-aws#150] [Ideas]
Should we automatically update versions.json based on triggered releases of APBS or PDB2PQR? For instance, there was a new release of PDB2PQR 3.5.0, should we just automatically update versions.json and start the build process?

It would be great to have the CI/CD include the APBS and PDB2PQR releases.

The risk is that the APBS or PDB2PQR dependencies change/break the web interface.

If we had sufficient testing of the web interface, we could do regression tests against the new releases and determine if an automatic update would be viable.

Without the tests, I think it is too risky to automatically update the versions and automatically deploy without humans overseeing the process.

**Comment by sobolevnrm** [comment]
Interesting!  GitHub actions support cron:  https://docs.github.com/en/actions/managing-issues-and-pull-requests/scheduling-issue-creation

**Reply by intendo** [reply]
We talked today about just having a cron action that built and tested everything once a day if anything had changed.

The biggest hurdle is getting the correct regression test suite that cover all the code in the web application, APBS, and PDB2PQR. This could be a huge effort.

**Reply by sobolevnrm** [reply]
It would be great just to have a cron-scheduled action to generate PDB2PQR and APBS `git tag -l | tail -1` output to compare with the current version and generate issues if the versions are different.

**Reply by Eo300** [reply]
I was experimenting yesterday/today to make a workflow to do achieve this (you probably saw many bot issues get generated this morning as a result).

This was all in the `playground_dest` branch for now, creating a script to grab the latest release, check if it's in the v**X**.**Y**.**Z** format, and if so create a new issue if the detected version is different.  The workflow/script files can be found here:
- https://github.com/Electrostatics/apbs-aws/blob/playground_dest/.github/workflows/create-issue-for-new-version.yml
- https://github.com/Electrostatics/apbs-aws/blob/playground_dest/.github/scripts/create_issue_for_new_version.sh

Latest workflow run: https://github.com/Electrostatics/apbs-aws/runs/4706925291?check_suite_focus=true
Example of generated issue (prior to me updating the contents): https://github.com/Electrostatics/apbs-aws/issues/156

If it looks ok to y'all I can turn it into a issue/PR

**Reply by sobolevnrm** [reply]
This looks great; thank you!

**Reply by intendo** [reply]
This looks great. We get the automation to create the issue and then the manual step to make sure we want to integrate the new release. Great work @Eo300
